Restructure oneDPL specification (#434)
oneapi-src/oneAPI-spec: source/elements/oneDPL/source/index.rst

The library is comprised of the following components:

- :doc:`Parallel API <parallel_api>`:

  * Parallel algorithms, complemented with execution
    policies and companion APIs for running on oneAPI devices.

  * An additional set of library classes and functions that
    are known to be useful in practice but are not yet included into
    C++ or SYCL specifications. 

- :doc:`SYCL Kernels API <sycl_kernels_api>`:

>   * A subset of the `C++ standard`_ library which you can use with

Reviewer: ```suggestion
  * A subset of the `C++ standard`_ library which can be used with
```

The reason is that the spec is a document primarily for implementers, not for users

Author: Applied suggestion
